controllers/nnf_access_controller_test.go

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2021, 2022 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import (
+	"context"
+	"reflect"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dwsv1alpha1 "github.com/HewlettPackard/dws/api/v1alpha1"
+	nnfv1alpha1 "github.com/NearNodeFlash/nnf-sos/api/v1alpha1"
+)
+
+var _ = Describe("Access Controller Test", func() {
+
+	nodeNames := []string{
+		"rabbit-nnf-access-test-node-1",
+		"rabbit-nnf-access-test-node-2"}
+
+	BeforeEach(func() {
+		for _, nodeName := range nodeNames {
+			ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nodeName}}
+			Expect(k8sClient.Create(context.TODO(), ns)).To(Succeed(), "Create Namespace")
+		}
+	})
+
+	Describe("Create Client Mounts", func() {
+
+		allocationNodes := make([]nnfv1alpha1.NnfStorageAllocationNodes, len(nodeNames))
+		for idx, nodeName := range nodeNames {
+			allocationNodes[idx] = nnfv1alpha1.NnfStorageAllocationNodes{
+				Count: 1,
+				Name:  nodeName,
+			}
+		}
+
+		It("Creates Lustre Client Mount", func() {
+
+			storage := &nnfv1alpha1.NnfStorage{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "nnf-access-test-storage",
+					Namespace: corev1.NamespaceDefault,
+				},
+				Spec: nnfv1alpha1.NnfStorageSpec{
+					FileSystemType: "lustre",
+					AllocationSets: []nnfv1alpha1.NnfStorageAllocationSetSpec{
+						{
+							Name: "MGTMDT",
+							NnfStorageLustreSpec: nnfv1alpha1.NnfStorageLustreSpec{
+								FileSystemName: "MGTMDT",
+								TargetType:     "MGTMDT",
+							},
+							Nodes: []nnfv1alpha1.NnfStorageAllocationNodes{
+								{
+									Count: 1,
+									Name:  corev1.NamespaceDefault,
+								},
+							},
+						},
+						{
+							Name: "OST",
+							NnfStorageLustreSpec: nnfv1alpha1.NnfStorageLustreSpec{
+								FileSystemName: "OST",
+								TargetType:     "OST",
+							},
+							Nodes: allocationNodes,
+						},
+					},
+				},
+				Status: nnfv1alpha1.NnfStorageStatus{
```

Comment:
You can remove this, I think.  It's deceptive, because it will be ignored by the Create().